torchci/scripts/check_alerts.py

```diff
@@ -0,0 +1,296 @@
+import json
+import os
+import urllib.parse
+from collections import defaultdict
+from curses.ascii import CAN
+from difflib import SequenceMatcher
+from email.policy import default
+from typing import Any, List, Tuple, Dict
+
+import requests
+
+HUD_API_URL = "https://hud.pytorch.org/api/hud/pytorch/pytorch/master/0"
+MAX_CONCURRENT_ALERTS = 1
+
+PENDING = "pending"
+NEUTRAL = "neutral"
+SKIPPED = "skipped"
+SUCCESS = "success"
+FAILURE = "failure"
+CANCELED = "canceled"
+
+ISSUES_WITH_LABEL_QUERY = """
+query ($owner: String!, $name: String!, $labels: [String!]) {
+  repository(owner: $owner, name: $name, followRenames: false) {
+    issues(first: 10, labels: $labels, states: [OPEN]) {
+      nodes {
+        id
+        title
+        closed
+        number
+      }
+    }
+  }
+}
+"""
+REPO_OWNER = "pytorch"
+REPO_NAME = "test-infra"
+failure_label = "pytorch-alert"
+labels = [failure_label]
+headers = {"Authorization": f"token {os.environ.get('GITHUB_TOKEN')}"}
+
+
+CREATE_ISSUE_URL = f"https://api.github.com/repos/{REPO_OWNER}/{REPO_NAME}/issues"
+UPDATE_ISSUE_URL = f"https://api.github.com/repos/{REPO_OWNER}/{REPO_NAME}/issues/"
+
+GRAPHQL_URL = "https://api.github.com/graphql"
+
+
+class JobStatus:
+    job_name: str = ""
+    jobs: List[Any] = []
+    current_status: str = None
+    job_statuses: List[Any] = []
+    failure_chain: List[Any] = []
+    flaky_jobs: List[Any] = []
+
+    def __init__(self, job_name: str, job_statuses: List[Any]):
+        self.job_name = job_name
+        self.job_statuses = job_statuses
+
+        filtered_statuses = list(filter(is_job_not_pending_or_skipped, job_statuses))
+        self.current_status = self.get_current_status(filtered_statuses)
+        self.failure_chain = self.get_most_recent_failure_chain(filtered_statuses)
+        self.flaky_jobs = self.get_flaky_jobs(filtered_statuses)
+
+    def get_current_status(self, job_statuses: List[Any]) -> str:
```

Comment:
Just curious, do you get a warning like "this function should be static etc .." for these methods?  I think you already assign `self.job_statuses = job_statuses`, so there is no need to pass `job_statuses` around in the parameter and `self.job_statuses` will do. This makes the code a bit shorter